docs/news/posts/2024-10-18-exploring-aiida-database.md

```diff
@@ -0,0 +1,339 @@
+Create a new text file and copy the content below (these are essentially python scripts, so you can use the `.py` extension).
+There are some comments which explain the purpose of each line of code, please read them carefully to understand how the query is constructed.
+If you are not familiar with the `QueryBuilder`, we refer to the [official AiiDA documentation for more details](https://aiida.readthedocs.io/projects/aiida-core/en/stable/howto/query.html).
+
+```python
+from aiida.orm import QueryBuilder, Dict, CalculationNode, BandsData, StructureData
+
+# Create a new query builder object
+qb = QueryBuilder()
+
+# I want, in the end, the 'band_gap' property returned ("projected")
+# This is in the attributes of the Dict node
+# As an additional challenge, I also want to filter them and get only those where the band gap (in eV) is < 0.5
+qb.append(
+    Dict,
+    project=['attributes.band_gap'],
+    filters={'attributes.band_gap': {'<': 0.5}},
+    tag='bandgap_node'
+)
+
+# This node must have been generated by a CalcFunctionNode (so, with outgoing link the node of the previously
+# part, that we tagged as `bandgap_node`, and I only want those where the
+# function name stored in the attributes is 'get_bandgap_inline'
+qb.append(
+    CalcFunctionNode,
+    filters={'attributes.function_name': 'get_bandgap_inline'},
+    with_outgoing='bandgap_node',
+    tag='bandgap_calc'
+)
+
+# One of the inputs should be a BandsData (band structure node in AiiDA)
+qb.append(BandsData, with_outgoing='bandgap_calc', tag='band_structure')
+
+# This should have been computed by a calculation (we know it's always Quantum ESPRESSO
+# in this specific DB, so I don't add more specific filters, but I could if I wanted to)
+qb.append(CalculationNode, with_outgoing='band_structure', tag='qe')
+
+# I want to get back the input crystal structure, and I want to get back
+# the whole AiiDA node (indicated with '*') rather than just some attributes
+qb.append(StructureData, with_outgoing='qe', project='*')
+
+# So, now, summarizing, I have decided to project on two things: the band_gap and the structure node.
+# I iterate on the query results, and I will get the two values for each matching result.
+for band_gap, structure in qb.all():
+    print("Band gap for {}: {:.3f} eV".format(structure.get_formula(), band_gap))
+```
+
+With these few lines of code (essentially 8, removing the comments, the indentation, and the import line) one is able to perform a query that will return all the structures (and band gaps) that are below a 0.5 eV threshold, but still with a finite gap (remeber, for metals, the current database sets `'band_gap': None`, as we saw above).
+You can now execute the script by running
+
+```bash
+$ verdi run <script_name>
+```
+
+Here is the output you should obtain if you only have the 2D materials database in your profile.
+
+```bash
+Band gap for I4Zr2: 0.416 eV
```

Comment:
I'd make this shorter, 5 entries or so should be enough.

Reply:
:+1: